zilliqa/src/consensus.rs

```diff
@@ -435,6 +447,37 @@ impl Consensus {
             .map(|opt| opt.unwrap_or_default())
     }
 
+    pub fn get_txns_to_execute(&mut self) -> Vec<SignedTransaction> {
+        // Note: push back Txns that have an incorrect nonce
+        self.new_transactions
+            .values()
+            .cloned()
+            .filter(|tx| {
+                if self.state.has_account(tx.from_addr)
+                    && self.state.must_get_account(tx.from_addr).nonce != tx.transaction.nonce
+                {
+                    // Incremement the value if it exists, otherwise set it to 1
+                    let retries = self.new_transactions_waiting.entry(tx.hash()).or_insert(0);
+                    let _ = retries.add(1);
+                    warn!(
+                        "Transaction nonce for tx {} is incorrect: {} when acct nonce is {}, retrying... {}/{}",
+                        tx.hash(),
+                        tx.transaction.nonce,
+                        self.state.must_get_account(tx.from_addr).nonce,
+                        retries,
+                        self.config.tx_retries
+                    );
+                    if *retries > self.config.tx_retries {
+                        warn!("Tranaction has exceeded retries and has been removed");
+                        self.new_transactions_waiting.remove(&tx.hash());
```

Comment:
Should it get removed from `self.new_transactions` here too? Otherwise it seems like next block, we'll just try to apply it again, resetting the retries counter back to 1 - unless I've missed something

Reply:
Yes -doh!

Reply:
can't mutate self.new_transactions while iterating over it so I just do it in a subsequent line, now